### PR TITLE
backupccl: skip TestDataDriven_user_defined_functions

### PR DESCRIPTION
**Pull request by msbutler**
Informs #124803

Release note: none

### CODE REVIEW

File: pkg/ccl/backupccl/testdata/backup-restore/user-defined-functions [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # Test backing up and restoring a database with user defined functions.
+skip issue-num=122353
```
Comment (rickystewart) [inline pkg/ccl/backupccl/testdata/backup-restore/user-defined-functions:2]: Should be 124803 instead, right?

Comment (msbutler) [reply pkg/ccl/backupccl/testdata/backup-restore/user-defined-functions:2]: it's the same issue.

**Review by rickystewart — APPROVED** [review]

**Comment by msbutler** [comment]
tftr!

bors r=rickystewart

**Comment by msbutler** [comment]
blathers backport 24.1

**Comment by yuzefovich** [comment]
blathers backport 24.1.1-rc

**Pull request merged**